com.microsoft.mrtk.graphicstools.unity/Runtime/Shaders/GraphicsToolsLighting.hlsl

```diff
@@ -26,6 +34,9 @@ GTMainLight GTGetMainLight()
     Light directionalLight = GetMainLight();
     light.direction = directionalLight.direction;
     light.color = directionalLight.color;
+#if defined(_RECEIVESHADOW)

```

Comment:
If light mode is _DISTANT_LIGHT I get this error:
Shader error in 'Graphics Tools/Standard': variable 'light' used without having been completely initialized....
You could move the this down, or add another clause to "#if defined(_DISTANT_LIGHT)"

Reply:
@pinkwerks 
Thank you!

I had completely overlooked DistantLight's behavior.
I fixed it to work for both DirectionalLight and DistantLight as #if defined(_RECEIVESHADOW) && defined(_URP).
How is this validation on Editor.


https://youtu.be/ci93t6u8CMc

Reply:
@pinkwerks 
Thanks for checking it out!
About the sample, I will try.
The best way I think to do it is to introduce the CastShadow feature as well. because that is a pair feature.

Currently, GraphicsToolsStandardShader has no functionality for shadows.
Therefore, in order to test the functionality currently submitted in PR, it is necessary to use it in combination with a shader that uses CastShadow, such as URP/Lit.

　My understanding is that it is not possible to compile branching by the Weber shader system on a per Pass basis.
　So if we want to add CastShadow to GraphicsToolsStandardShader, there is a way to build another shader that shares the hlsl file with a name like GraphicsToolsStandard"CastShadow".shader etc. I think.
　Which method do you think is best?

Reply:
Thank you for explaining. My problem above was I forgot to enable shadows on the camera 😳. I did see the gt-standard shaders receiving shadows thanks to your PR!

I spoke with @Cameron-Micka and we both think adding shadow casting is important. We understand there may be a cost with adding another Pass to the existing shaders... so we're going to do some investigation on the impact of adding more keywords and passes.

For this PR we won't consider casting shadows to keep it simple. Thanks again!